src/dune_rules/gen_rules.ml

```diff
@@ -64,9 +62,11 @@ end = struct
   let of_stanza stanza ~sctx ~src_dir ~ctx_dir ~scope ~dir_contents ~expander
       ~files_to_install =
     let dir = ctx_dir in
+    let toplevel_setup = Toplevel.Stanza.setup in
+    let open Dune_file in
     match stanza with
     | Toplevel toplevel ->
-      let+ () = Toplevel_rules.setup ~sctx ~dir ~toplevel in
+      let+ () = toplevel_setup ~sctx ~dir ~toplevel in
```

Comment:
if `Topleve.Stanza.setup` is only used here, do we need the `let toplevel_setup = Toplevel.Stanza.setup` expression?

Reply:
We need it because `let open Dune_file` shadows the `Toplevel` module.